instance/models/mixins/openedx_monitoring.py

```diff
@@ -44,33 +45,92 @@ def enable_monitoring(self):
             return
         self.logger.info('Checking New Relic Synthetics monitors')
 
-        urls_to_monitor = self._urls_to_monitor  # Store locally so we don't keep re-computing this
+        # The notifications for the monitors are now set up using New Relic alert policies. Alert policies
+        # can contain one or more alert conditions and an alert condition can be associated with
+        # one or more Synthetics (aka availability) monitors.
+        #
+        # An alert policy can have one or more existing notification channels added to it. We use email
+        # notification channels for sending out email notifications.
+        #
+        # The alert policy can be configured to create an incident per policy, per alert condition or per monitor and
+        # entity (for each failure). We use the 'per-policy' setting on all the alert policies.
+
+        try:
+            alert_policy = NewRelicAlertPolicy.objects.get(instance=self)
+        except NewRelicAlertPolicy.DoesNotExist:
+            alert_policy_id = newrelic.add_alert_policy(self.domain)
+            alert_policy = NewRelicAlertPolicy.objects.create(id=alert_policy_id, instance=self)
+
+        urls_to_monitor_dict = self._urls_to_monitor
+        urls_to_monitor = set(urls_to_monitor_dict)  # Store locally so we don't keep re-computing this
         already_monitored_urls = set()
 
         for monitor in self.new_relic_availability_monitors.all():
             url = newrelic.get_synthetics_monitor(monitor.pk)['uri']
             if url in urls_to_monitor:
                 already_monitored_urls.add(url)
+                # Check if the monitor has an associated alert condition
+                # and create one if it doesn't. This helps when the alert condition
+                # wasn't created in a previous invocation due to some issue.
+                if not monitor.new_relic_alert_conditions.exists():
```

Comment:
This seems to take care of previously provisioned instances that used the old Synthetics. :+1:

Reply:
@giovannicimolin, this also takes care of the scenario where the alert policy creation failed in a run and we retry.

Reply:
@giovannicimolin, I have added a unit test to verify that this works for the instances that already have the monitoring enabled using the old code.